frontend/main.js

```diff
@@ -103,19 +106,45 @@ async function startApp () {
     // Since a runtime error just occured, we likely want to persist app logs to local storage now.
     sbp('appLogs/save')
   }
+
+  await sbp('gi.db/settings/load', 'CHELONIA_STATE').then(async (cheloniaState) => {
+    // TODO: PLACEHOLDER TO SIMULATE CHELONIA IN A SW
+    if (!cheloniaState) return
+    const identityContractID = await sbp('gi.db/settings/load', SETTING_CURRENT_USER)
+    if (!identityContractID) return
+    Object.assign(sbp('chelonia/rootState'), cheloniaState)
+  })
+
+  const saveChelonia = () => sbp('okTurtles.eventQueue/queueEvent', 'CHELONIA_STATE', () => {
+    return sbp('gi.db/settings/save', 'CHELONIA_STATE', sbp('chelonia/rootState'))
+  })
+  const saveCheloniaDebounced = debounce(saveChelonia, 200)
+
   await sbp('chelonia/configure', {
     connectionURL: sbp('okTurtles.data/get', 'API_URL'),
-    stateSelector: 'state/vuex/state',
-    reactiveSet: Vue.set,
-    reactiveDel: Vue.delete,
+    // stateSelector: 'state/vuex/state',
+    reactiveSet: (o: Object, k: string, v: string) => {
+      // Simulate SW environment
+      if (o[k] !== v) {
+        o[k] = v
+        saveCheloniaDebounced()
+      }
+    },
+    reactiveDel: (o: Object, k: string) => {
+      // Simulate SW environment
+      if (has(o, k)) {
```

Comment:
I think this `if (has..` thing can be removed. No need for an extra branch here

Reply:
Same as for `reactiveSet`. This is for performance reasons.

Reply:
If I had to guess, I'd bet that in most cases the call to `reactiveDel` deletes an actual key-value mapping. If that's true, then in most cases this check is unnecessary and therefore serves to slow things down unnecessarily.